src/plugins/slurm/slurm-spank-auks.c

```diff
@@ -589,31 +587,33 @@ spank_auks_remote_exit (spank_t sp, int ac, char **av)
 	if ( setegid(gid) ) {
 		xerror("unable to switch to user gid : %s",
 		      strerror(errno));
-		return (-1);
+		fstatus = -1;
+		goto err;
 	}
 
 	/* change to user uid and gid before removing cred */
 	if ( seteuid(uid) ) {
 		xerror("unable to switch to user uid : %s",
 		      strerror(errno));
-		setegid(getgid());
-		return (-1);
+		fstatus = -1;
+		goto err;
 	}
 
 	/* sync all/some file systems to ensure dirty pages flush
 	   while we are sure to still have a ticket to do that
 	   (see _sync_fs method for more details) */
 	_sync_fs();
 
-	/* remove credential cache */
-	fstatus = unlink(auks_credcache);
-	if ( fstatus != 0 ) {
-		xerror("unable to remove user '%u' cred cache %s : %s",
-		      uid,auks_credcache,strerror(errno));
+	/* Destroy all krb5 ccache */
+	fstatus = auks_krb_cc_destroy(auks_credcache);
+	if (fstatus) {
+	      xerror("Unable to destroy ccache %s",auks_credcache);
+	      goto err;
 	}
-	else
-		xinfo("user '%u' cred cache %s removed",uid,auks_credcache);
 
+        xinfo("Destroyed ccache %s", auks_credcache);
+
```

Comment:
this is probably here that you should free auks_credcache and set it back to NULL.